packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -3557,6 +3560,11 @@ class InputDecoration {
   /// a default height based on text size.
   final BoxConstraints? constraints;
 
+  /// Whether the hint text should be adapt its height when the user enters text.
```

Comment:
Maybe: 

Whether the input field's height should always be greater than or equal to the height of the hint text, even if the hint text is not visible.

This is useful when [hintText] is ...... 

Defaults to false, which allows the input field to ignore the height of the [hintText] during layout when it is not displayed.

Reply:
also consider updating the field name accordingly

Reply:
Thank you for your review, I got it